app/population/middleware/set-population.js

```diff
@@ -0,0 +1,58 @@
+const locationsFreeSpacesService = require('../../../common/services/locations-free-spaces')
+const moveService = require('../../../common/services/move')
+const populationService = require('../../../common/services/population')
+
+async function setPopulation(req, res, next) {
+  try {
+    const { locationId, date } = req.params
+    let transfersIn = 0
+    let transfersOut = 0
+
+    const dailyFreeSpace = await locationsFreeSpacesService.getPrisonFreeSpaces(
+      {
+        dateFrom: date,
+        dateTo: date,
+        locationIds: locationId,
+      }
+    )
+
+    const freeSpacePopulation =
+      dailyFreeSpace?.[0]?.meta?.populations?.[0] || {}
+    const { id: populationId } = freeSpacePopulation
+
+    if (populationId) {
+      req.population = await populationService.getById(populationId, {
+        include: ['moves_from', 'moves_to', 'location'],
+      })
+
+      transfersIn = req.population.moves_to.length
+      transfersOut = req.population.moves_from.length
+    }
+
+    if (!req.population) {
+      ;[transfersIn, transfersOut] = await Promise.all([
+        moveService.getActive({
```

Comment:
You could extract the shared options here to reduce duplication: 

```javascript
const options = { dateRange: [date, date], isAggregation: true }
const [transfersIn, transfersOut] = await Promise.all([
  moveService.getActive({ ...options, toLocationId: locationId }),
  moveService.getActive({ ...options, fromLocationId: locationId }),
])
```

Reply:
Shared options extracted.